crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1187,24 +1191,100 @@ pub struct RpcServerConfig {
     ipc_endpoint: Option<String>,
     /// JWT secret for authentication
     jwt_secret: Option<JwtSecret>,
+    /// Additional middleware layer.
+    additional_middleware: Option<L>,
```

Comment:
this is the right approach to configure the service for all _servers_

if we look at the `ServerBuilder::set_http_middleware` function then this accepts a `http_middleware: tower::ServiceBuilder<T>,`

so we can do the same here, starting with `tower::ServiceBuilder<L>,` then we also don't need to restrict the `L` generic above